### DISCUSSION TITLE
Wrapper for TableNode

### DISCUSSION THREAD

**Discussion by richardaum** [facebook/lexical#6669] [Q&A]
Hi! This is my first time here, and first of all I'd like to thank you all for the incredible work here.

So here I am, trying to build a wrapper for a table so I can customize it adding extra features, cool behavior, etc. The way I thought of it was creating a `TableWrapper `as a `ElementNode` with a `relative` position, so the elements inside the table can be positioned properly:

```
[TableWrapper extends ElementNode] > 
  [
    CustomTableNode extends TableNode,
    AbsoluteElement1 extends DecoratorNode, 
    AbsoluteElement2 extends DecoratorNode
  ] > 
    etc.
```

By creating the wrapper as an `ElementNode`, it means it is selectable, specially when navigating using the keyboard, which breaks the keyboard navigation that's already implemented in Lexical, making things like being able to add text directly inside the wrapper. I'd like to Lexical think of the `TableWrapper `as an "unselectable element" but still have `Node` children - so when it's selected using the keyboard when navigating up it selects its previous sibling instead of the table's one.

I've tried to play around my own keyboard listeners and I could overcome going from inside to outside the table by using some code from `LexicalTableSelectionHelpers` functions like a changed version of those `isExitingTableAnchor` and `$handleTableExit`. But when it comes to going from outside to inside the table, things get complicated which makes the same logic to not work.

By the way, I took a look on how a similar feature is made on Playground, but it's relying on `mousemove`, bounding rects, x/y... **I'm looking for something more close to a relative/absolute approach** without relying on tracking the TableNode look for they actual positioning by using JS.

That journey made me think, maybe I'm overcomplicating this implementation and maybe there is another thing I could do to my `TableWrapper` to make it unselectable/skipable/invisible to Lexical.

Thoughts?

**Comment by liuweiGL** [comment]
I have also encountered this problem. May I ask if you have found a solution to it?
